saos-restful-web-api/src/test/java/pl/edu/icm/saos/api/dump/enrichmenttag/DumpEnrichmentTagControllerTest.java

```diff
@@ -0,0 +1,199 @@
+        
+    }
+    
+    @Test
+    public void showEnrichmentTags_PAGING() throws Exception {
+        
+        // given
+        
+        int pageSize = 2;
+        int pageNumber = 1;
+        
+        
+        // execute
+        
+        ResultActions result = mockMvc.perform(get(DUMP_ENRICHMENT_PATH)
+                .param(PAGE_SIZE, String.valueOf(pageSize))
+                .param(PAGE_NUMBER, String.valueOf(pageNumber))
+                .accept(MediaType.APPLICATION_JSON));
+        
+        
+        // assert
+        
+        result
+            .andExpect(jsonPath("$.items", hasSize(1)))
+            
+            .andExpect(jsonPath("$.items.[0].id").value(enrichmentTags.get(2).getId()))
+            .andExpect(jsonPath("$.items.[0].judgmentId").value(enrichmentTags.get(2).getJudgmentId()))
+            .andExpect(jsonPath("$.items.[0].tagType").value("KEYWORDS"))
+            .andExpect(jsonPath("$.items.[0].value.keywords.[0]").value("val11"))
+            .andExpect(jsonPath("$.items.[0].value.keywords.[1]").value("val21"))
+            .andExpect(jsonPath("$.items.[0].value.keywords.[2]").value("val32"))
+
+            .andExpect(jsonPath("$.queryTemplate.pageSize.value").value(pageSize))
+            .andExpect(jsonPath("$.queryTemplate.pageNumber.value").value(pageNumber));
+    }
+    
+    @Test
+    public void showEnrichmentTags_WRONG_PARAM() throws Exception {
+        
+        // execute
+        
+        ResultActions actions = mockMvc.perform(get(DUMP_ENRICHMENT_PATH)
+                .param("some_incorrect_parameter_name", "")
+                .accept(MediaType.APPLICATION_JSON));
+
+        
+        // assert
+        
+        actions.andExpect(status().isBadRequest());
+    }
+    
+    
+    //------------------------ PRIVATE --------------------------
+    
+    private void createEnrichmentTags() {
+        EnrichmentTag enrichmentTag;
+        
+        String jsonValue1 = normalizeJson("{'ref':'AAA1'}");
+        String jsonValue2 = normalizeJson("{'caseNumbers':['XYZ1','XYZ2']}");
+        String jsonValue3 = normalizeJson("{'keywords':['val11','val21','val32']}");
+        
+        enrichmentTag = createEnrichmentTag(testObjectContext.getCcCourtId(), "REFERENCED_REGULATIONS", jsonValue1);
+        enrichmentTags.add(enrichmentTag);
+        enrichmentTag = createEnrichmentTag(testObjectContext.getCcCourtId(), "REFERENCED_CASE_NUMBERS", jsonValue2);
+        enrichmentTags.add(enrichmentTag);
+        enrichmentTag = createEnrichmentTag(testObjectContext.getCcCourtId(), "KEYWORDS", jsonValue3);
+        enrichmentTags.add(enrichmentTag);
+        
+    }
+    
+    private EnrichmentTag createEnrichmentTag(int judgmentId, String tagType, String jsonValue) {
+        EnrichmentTag enrichmentTag = new EnrichmentTag();
+        
+        enrichmentTag.setJudgmentId(judgmentId);
+        enrichmentTag.setTagType(tagType);
+        enrichmentTag.setValue(jsonValue);
```

Comment:
I think it would be nice to create a TestInMemoryUploadEnrichmentTagFactory equivalent for EnrichmentTags and perhaps PersistentUploadEnrichmentTagFactory.